pkg/remote/resolution/error.go

```diff
@@ -21,55 +21,63 @@ import (
 	"fmt"
 )
 
-// ErrorRequestedResourceIsNil is returned when remote resolution
-// appears to have succeeded but the resolved resource is nil.
-var ErrorRequestedResourceIsNil = errors.New("unknown error occurred: requested resource is nil")
+var (
+	// ErrNilResource is returned when remote resolution
+	// appears to have succeeded but the resolved resource is nil.
+	ErrNilResource = errors.New("unknown error occurred: requested resource is nil")
+
+	// ErrorRequestedResourceIsNil is a deprecated alias for ErrNilResource and will
+	// be removed in a future release.
+	//
+	// Deprecated: use ErrNilResource instead.
+	ErrorRequestedResourceIsNil = ErrNilResource
+)
 
-// ErrorInvalidRuntimeObject is returned when remote resolution
+// InvalidRuntimeObjectError is returned when remote resolution
 // succeeded but the returned data is not a valid runtime.Object.
-type ErrorInvalidRuntimeObject struct {
+type InvalidRuntimeObjectError struct {
```

Comment:
I think we need similar backwards compatible [aliasing](https://go.dev/ref/spec#Type_identity) for these types as well - e.g.

```go
// ErrorInvalidRuntimeObject is returned when remote resolution
// succeeded but the returned data is not a valid runtime.Object.
// Deprecated: ...
type ErrorInvalidRuntimeObject = InvalidRuntimeObjectError
```

Reply:
Ugh, this gets a big `doh` from me... I addressed the specific comments where this issue was raised but I didn't go through the other changes. Combing through now...

Reply:
I think the type aliases for the struct errors are still missing (maybe need to push?)

Reply:
Right you are, good catch!